### PR TITLE
[C-1064] Show "You're Offline" component on mobile

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description
Screen Shot 2022-09-16 at 11 03 47 AM

On mobile, replace screens with a "You're Offline" component instead of trying and failing to show content when the device is not connected to the internet.
_Note: In the future, the favorites screen will show offline downloads, but we're showing this until that feature is implemented._

Removes the `ReachabilityBanner` component on mobile.

This is not a change to how we check whether we're connected, just what we show to the user.

https://www.loom.com/share/bf6b4373b1294f0699e46c00aadb90ee

### How Has This Been Tested?

Tested on simulator and physical iOS device. Toggling wifi on and off triggered the screen change, and reconnecting returned to the content screens as expected in all tested cases.

### How will this change be monitored?

N/A

### Feature Flags ###

We will have a feature flag for Offline Mode, but this is separable and usable today, so we will just ship it.

### CODE REVIEW

File: packages/mobile/src/components/offline-placeholder/OfflinePlaceholder.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const { unboxed } = props
+  const styles = useStyles()
+  const [isRefreshing, setIsRefreshing] = useState(false)
+
+  const handleRefresh = useCallback(() => {
+    if (isRefreshing) return
+    setIsRefreshing(true)
+    // NetInfo.refresh() usually returns almost instantly
+    // Introduce minimum wait to convince user we took action
+    Promise.all([NetInfo.refresh(), wait(getRandomInt(800, 1200))]).then(() =>
```
Comment (sliptype) [inline packages/mobile/src/components/offline-placeholder/OfflinePlaceholder.tsx:58]: Nice! Maybe we just keep it simple and get rid of the randomness though?

File: packages/mobile/src/screens/profile-screen/ProfileHeader/ProfileHeader.tsx [hunk 2]
```diff
@@ -109,6 +115,19 @@ export const ProfileHeader = (props: ProfileHeaderProps) => {
     LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
   }, [isExpanded, setIsExpanded])
 
+  if (isNotReachable) {
```
Comment (sliptype) [inline packages/mobile/src/screens/profile-screen/ProfileHeader/ProfileHeader.tsx:118]: Can this check be nested further in the jsx? So we aren't duplicating 
```ts
        <CoverPhoto scrollY={scrollY} />
        <ProfilePicture style={styles.profilePicture} />
        <View pointerEvents='box-none' style={styles.header}>
          <ProfileInfo onFollow={handleFollow} />
...
```

**Review by sliptype — APPROVED** [review]
Awesome stuff! Thanks for making those doc updates too

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/amendel-ur-offline

**Review by dylanjeffers — APPROVED** [review]
Awesome job, code is real clean too

File: packages/mobile/src/components/offline-placeholder/OfflinePlaceholder.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+type OfflinePlaceholderProps = {
+  unboxed?: boolean
+}
+
+export const OfflinePlaceholder = (props: OfflinePlaceholderProps) => {
+  const { unboxed } = props
+  const styles = useStyles()
+  const [isRefreshing, setIsRefreshing] = useState(false)
+
+  const handleRefresh = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/mobile/src/components/offline-placeholder/OfflinePlaceholder.tsx]: ooh neat!

File: packages/mobile/src/screens/favorites-screen/FavoritesScreen.tsx [hunk 4]
```diff
@@ -30,11 +35,16 @@ const favoritesScreens = [
...
 export const FavoritesScreen = () => {
   usePopToTopOnDrawerOpen()
+  const isNotReachable = useSelector(getIsReachable) === false
 
   return (
     <Screen>
       <Header text='Favorites' />
-      <TopTabNavigator screens={favoritesScreens} />
+      {isNotReachable ? (
+        <OfflinePlaceholder />
```
[the remark this answers is not in the document]
Comment (amendelsohn) [reply packages/mobile/src/screens/favorites-screen/FavoritesScreen.tsx]: Omg. I can't believe I didn't do this already lol. I knew I was repeating myself a lot 🤦

SOO much cleaner now. Ty!

**Pull request merged**